### PR TITLE
refactor(query): fix create as take database as catalog

### LINKED ISSUE #7457: bug: where clause error
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

8.0.26-v0.8.12-nightly-74d0287-simd(rust-1.64.0-nightly-2022-08-27T03:09:50.519081067Z)

### What's Wrong?

https://databend.rs/doc/reference/data-types/data-type-logical-types#implicit-conversion

negative numbers are not converted to true

### How to Reproduce?

```sql
select * from t0 where (-123);
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### LINKED ISSUE #7496: bug: create table x like tb1;  tb1 will find in default database
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

```
select version();
+----------------------------------------------------------------------------------------+
| version()                                                                              |
+----------------------------------------------------------------------------------------+
| 8.0.26-v0.8.25-nightly-6a5d857-simd(rust-1.64.0-nightly-2022-09-04T16:37:49.86847381Z) |
+----------------------------------------------------------------------------------------+
1 row in set (0.002 sec)
```

### What's Wrong?

```
create database wubx; 
use wubx;
create table w_tb1(id int)
MySQL [wubx]> create table w_tb2 like w_tb1;
ERROR 1105 (HY000): Code: 1025, displayText = Unknown table 'w_tb1'.
```

### How to Reproduce?

above

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### LINKED ISSUE #7498: bug: cannot convert NULL to a non-nullable type
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

8.0.26-v0.8.26-nightly-c8c59c1-simd(rust-1.64.0-nightly-2022-09-05T16:29:30.95458078Z)

### What's Wrong?

cannot convert NULL to a non-nullable type

### How to Reproduce?

```sql
create table t0(c0 double null, c1 double null);
insert into t0 values(NULL,0.0),(NULL,0.0),(NULL,0.0);

MySQL [sqlancer]> select c0::float32 from t0;
+-------------+
| c0::Float32 |
+-------------+
|         0.0 |
|         0.0 |
|         0.0 |
+-------------+
3 rows in set (0.006 sec)
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1. fix create as take database as catalog
2. fix number cast into boolean bug
3. throw an error when cast nullable data into non-nullable

Fixes #7457
Fixes #7496
Fixes #7498

### CODE REVIEW

File: tests/logictest/suites/base/05_ddl/05_0000_ddl_create_tables [hunk 1]
```diff
@@ -164,24 +164,24 @@ statement query TTTTT
...
-double DOUBLE NO 0 
-date DATE NO 0 
-datetime TIMESTAMP(6) NO 0 
-ts TIMESTAMP(6) NO 0 
-str VARCHAR NO 3 
-bool BOOLEAN NO false 
-arr ARRAY NO [] 
-obj OBJECT NO {} 
-variant VARIANT NO null 
+tiny TINYINT NO 0
```
Comment (sundy-li) [inline tests/logictest/suites/base/05_ddl/05_0000_ddl_create_tables:167]: my editor will auto strip the trailing spaces...

I think it's bad test case.

**Review by b41sh — APPROVED** [review]
LGTM

File: tests/logictest/suites/base/03_dml/03_0005_select_filter [hunk 2]
```diff
@@ -23,6 +24,12 @@ SELECT number as c1, (number+1) as c2 FROM numbers_mt (3) where number+1>1;
 1 2
 2 3
 
+statement query I
+SELECT count() FROM numbers_mt (10) where -1;
```
Comment (leiysky) [inline tests/logictest/suites/base/03_dml/03_0005_select_filter:28]: Shall we allow this implicit conversion?

Comment (sundy-li) [reply tests/logictest/suites/base/03_dml/03_0005_select_filter:28]: Yes, we allow this only in the `filter`.

**Review by xudong963 — APPROVED** [review]

**Pull request merged**